### PR TITLE
Add JSON

### PR DESCRIPTION
**Pull request by hanachin**

### CODE REVIEW

File: ruby-signature.gemspec [hunk 1]
```diff
@@ -42,4 +42,5 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "rubocop"
   spec.add_development_dependency "rubocop-rubycw"
   spec.add_development_dependency "minitest-reporters", "~> 1.3.6"
+  spec.add_development_dependency "json", "~> 2.3.0"
```
Comment (soutaro) [inline ruby-signature.gemspec:45]: Okay, JSON is a gem... I think the best way is to put RBS in JSON gem, but I think it's okay to have the signatures here for now.

File: stdlib/builtin/builtin.rbs [hunk 2]
```diff
@@ -37,3 +37,4 @@ type int = Integer | _ToInt
 type real = Integer | Float | Rational
 
 type string = String | _ToStr
+type encoding = Encoding | string
```
Comment (soutaro) [inline stdlib/builtin/builtin.rbs:40]: 👍

File: stdlib/json/json.rbs [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+  #
+  def self.dump_default_options: () -> json_options
+
+  def self.dump_default_options=: (json_options) -> json_options
+
+  def self.fast_generate: (_ToJson obj, ?json_options opts) -> String
+
+  alias self.fast_unparse self.fast_generate
+
+  def self.generate: (_ToJson obj, ?json_options opts) -> String
```
Comment (soutaro) [inline stdlib/json/json.rbs:143]: Could you rewrite these module functions using `def self?.generate: ...` syntax?

Comment (hanachin) [reply stdlib/json/json.rbs]: Thanks for the review! I updated the signature to use the `def self?.` syntax.

**Comment by hanachin** [comment]
@soutaro Done :v:

**Pull request merged**